src/InstrumentationEngine.Lib/Logging.cpp

```diff
@@ -217,7 +217,28 @@ void CLogging::XmlDumpHelper::WriteStringNode(const WCHAR* name, const WCHAR* va
     {
         return;
     }
-    m_result << m_childrenIndent << _T("<") << name << _T(">") << value << _T("</") << name << _T(">") << _T("\r\n");
+    m_result += m_childrenIndent + _T("<") + name + _T(">") + value + _T("</") + name + _T(">") + _T("\r\n");
+}
+
+namespace {
+    tstring format_hex(ULONG32 value, size_t width = 10)
```

Comment:
>width = 10 [](http://example.com/codeflow?start=45&length=10)

This width is a little unintuitive. It doesn't represent zero padding, it represents the size of the string. I would expect it to represent the padding.

Reply:
The parameter is never used anyway. Consider removing it so that the underflow issue below can't happen.

Reply:
Good point. I was mimicking setw from the stringstream.